core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java

```diff
@@ -265,9 +265,20 @@ private synchronized void loadStoredKeysAndFileNames() throws ObjectStoreExcepti
             File[] files = listValuesFiles();
             for (int i = 0; i < files.length; i++)
             {
-                File file = files[i];
-                StoreValue<T> storeValue = deserialize(file);
-                realKeyToUUIDIndex.put(storeValue.getKey(), file.getName());
+                try
+                {
+                    File file = files[i];
+                    StoreValue<T> storeValue = deserialize(file);
+                    realKeyToUUIDIndex.put(storeValue.getKey(), file.getName());
+                }
+                catch (ObjectStoreException e)
+                {
+                    if (logger.isWarnEnabled())
+                    {
+                        logger.warn("Could not deserialize an ObjectStore file");
```

Comment:
why not moving the file to a special directory instead? if you delete it we'll never have any tools to try to diagnose or have a sense of why the file is corrupted. Also, at the customer's eyes, moving the file is less violent than deleting it